### PR TITLE
Show logs from a selected target in the editor's console

### PR DESCRIPTION
**Pull request by AGulev**
When a remote target is selected, the editor tries to connect to the target's log service to display them in the console.

Fix https://github.com/defold/defold/issues/6911
## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [x] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [x] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [x] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [x] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [x] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: editor/src/clj/editor/app_view.clj [hunk 1]
```diff
@@ -630,32 +630,6 @@
     (build-errors-view/update-build-errors build-errors-view error-value)
     (show-build-errors! main-scene tool-tab-pane)))
 
-(def ^:private remote-log-pump-thread (atom nil))
```
Comment (AGulev) [inline editor/src/clj/editor/app_view.clj:633]: Console relater code moved to `console/` to be able to use it from both `targets` and `app-view`

File: editor/src/clj/editor/app_view.clj [hunk 2]
```diff
@@ -801,23 +775,14 @@
...
-
 (defn- make-launched-log-sink [launched-target on-service-url-found]
   (let [initial-output (atom "")]
     (fn [line]
       (when (< (count @initial-output) 5000)
         (swap! initial-output str line "\n")
         (when-let [target-info (engine/parse-launched-target-info @initial-output)]
           (targets/update-launched-target! launched-target target-info on-service-url-found)))
-      (when (= @console-stream (:log-stream launched-target))
+      (when (console/current-stream? (:log-stream launched-target))
```
Comment (AGulev) [inline editor/src/clj/editor/app_view.clj:785]: new function to check stream

File: editor/src/clj/editor/console.clj [hunk 3]
```diff
@@ -93,6 +94,51 @@
   []
   (swap! pending-atom assoc :clear true :entries [] :index 0))
 
+(def ^:private remote-log-pump-thread (atom nil))
+(def ^:private console-stream (atom nil))
+
+(defn current-stream? [stream] (= @console-stream stream))
```
Comment (AGulev) [inline editor/src/clj/editor/console.clj:100]: this code moved from `app-veiw` except `current-stream?` function. This is a new one

File: editor/src/clj/editor/engine.clj [hunk 4]
```diff
@@ -130,27 +129,28 @@
...
-            ;; Setting to 0 means wait indefinitely for new data
-            (.setSoTimeout socket 0)
-            is)
-          (do
-            (.close socket)
-            nil)))
-      (catch Exception e
-        (.close socket)
-        (throw e)))))
+  (when (and (:log-port target) (:address target))
```
Comment (AGulev) [inline editor/src/clj/editor/engine.clj:132]: the only change in this block is this line.

File: editor/src/clj/editor/targets.clj [hunk 5]
```diff
@@ -321,9 +324,22 @@
...
+        {:type :error
+         :id ::target-connection-error
+         :text msg}))))
+
 (defn select-target! [prefs target]
   (reset! selected-target-atom target)
   (prefs/set-prefs prefs "selected-target-id" (:id target))
+  (let [log-stream (engine/get-log-service-stream target)]
+    (when log-stream
+      (console/set-log-service-stream log-stream)))
```
Comment (AGulev) [inline editor/src/clj/editor/targets.clj:342]: This one line is the main change of the commit

File: editor/src/clj/editor/targets.clj [hunk 6]
```diff
@@ -321,9 +324,22 @@
...
 
+(defn- show-error-message [exception workspace]
+  (ui/run-later
+    (let [msg (str (.getMessage exception) "\n\n"
+                   "The target you have chosen isn't available")]
+      (notifications/show!
+        (workspace/notifications workspace)
+        {:type :error
+         :id ::target-connection-error
+         :text msg}))))
```
Comment (AGulev) [inline editor/src/clj/editor/targets.clj:335]: When connection is impossible, show a notification instead of java exception

**Comment by AGulev** [comment]
It works with mobile devices as well:
https://github.com/user-attachments/assets/9fca3989-9437-4eda-bced-879e3ad34aa4

**Review by matgis — APPROVED** [review]
A really nice improvement! LGTM! ❤️

**Pull request merged**